tools/clang/unittests/SPIRV/WholeFileCheck.h

```diff
@@ -0,0 +1,335 @@
+    flags.push_back(L"-T");
+    flags.push_back(profile.c_str());
+    flags.push_back(L"-spirv");
+
+    IFT(dllSupport.CreateInstance(CLSID_DxcLibrary, &pLibrary));
+    IFT(pLibrary->CreateBlobFromFile(srcFile.c_str(), nullptr, &pSource));
+    IFT(pLibrary->CreateIncludeHandler(&pIncludeHandler));
+    IFT(dllSupport.CreateInstance(CLSID_DxcCompiler, &pCompiler));
+    IFT(pCompiler->Compile(pSource, srcFile.c_str(), entry.c_str(),
+                           profile.c_str(), flags.data(), flags.size(), nullptr,
+                           0, pIncludeHandler, &pResult));
+    IFT(pResult->GetStatus(&resultStatus));
+
+    if (SUCCEEDED(resultStatus)) {
+      CComPtr<IDxcBlobEncoding> pStdErr;
+      IFT(pResult->GetResult(&pCompiledBlob));
+      convertIDxcBlobToUint32(pCompiledBlob);
+      success = true;
+    } else {
+      IFT(pResult->GetErrorBuffer(&pErrorBuffer));
+      fprintf(stderr, "%s\n", (char *)pErrorBuffer->GetBufferPointer());
+      success = false;
+    }
+  } catch (...) {
+    // An exception has occured while running the compiler with SPIR-V
+    // Generation
+    success = false;
+  }
+
+  return success;
+}
+
+bool WholeFileTest::disassembleSpirvBinary(bool generateHeader) {
+  uint32_t options = SPV_BINARY_TO_TEXT_OPTION_FRIENDLY_NAMES;
+  if (!generateHeader)
+    options |= SPV_BINARY_TO_TEXT_OPTION_NO_HEADER;
+  return spirvTools.Disassemble(generatedBinary, &generatedSpirvString,
+                                options);
+}
+
+bool WholeFileTest::ValidateSpirvBinary() {
+  return spirvTools.Validate(generatedBinary);
+}
+
+void WholeFileTest::convertIDxcBlobToUint32(CComPtr<IDxcBlob> &blob) {
+  size_t num32BitWords = (blob->GetBufferSize() / 4);
+  if (blob->GetBufferSize() % 4 > 0)
+    num32BitWords++;
+  std::string binaryStr((char *)blob->GetBufferPointer(),
+                        blob->GetBufferSize());
+  binaryStr.resize(num32BitWords * 4, 0);
+  generatedBinary.resize(num32BitWords, 0);
+  for (size_t i = 0; i < num32BitWords; ++i) {
+    generatedBinary[i] =
+        (reinterpret_cast<const uint32_t *>(binaryStr.data()))[i];
+  }
+}
+
+bool WholeFileTest::compareExpectedSpirvAndGeneratedSpirv() {
+  return generatedSpirvString == expectedSpirvString;
+}
+
+std::string WholeFileTest::getAbsPathOfInputDataFile(std::string filename) {
+  std::string path = clang::spirv::testOptions::inputDataDir;
+  char sep = '/';
+
+#if defined(_WIN32)
+  sep = '\\';
+  std::replace(path.begin(), path.end(), '/', '\\');
+#endif
+
+  if (path[path.size() - 1] != sep) {
+    path = path + sep;
+  }
+  path += filename;
+  return path;
+}
+
+bool WholeFileTest::runWholeFileTest(std::string filename, bool generateHeader,
+                                     bool runSpirvValidation) {
+  inputFilePath = getAbsPathOfInputDataFile(filename);
+  hlslSourceFilePath = inputFilePath + hlslSuffix;
+
+  bool success = true;
+
+  // Parse the input file.
+  success = success && parseInputFile();
+
```

Comment:
There's no need to explicitly check if `success` is already false because:
if you have `A && B` , `B` is not executed unless A is true.